packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -129,6 +128,10 @@ export function getValidatorApi({
   const MAX_API_CLOCK_DISPARITY_SEC = Math.min(0.5, config.SECONDS_PER_SLOT / 2);
   const MAX_API_CLOCK_DISPARITY_MS = MAX_API_CLOCK_DISPARITY_SEC * 1000;
 
+  let defaultGraffiti: string = "";
+  void updateDefaultGraffiti();
+  chain.clock.on(ClockEvent.epoch, updateDefaultGraffiti);
```

Comment:
I don't really like this approach, for two reasons
- we run too many tasks already on epoch clock event
- we don't need to update this every epoch

What about running this prepare slot but only we are proposer, e.g. after emitting payload attributes
https://github.com/ChainSafe/lodestar/blob/b453b37d53a946b6faf73277fe1d75becff3ae8f/packages/beacon-node/src/chain/prepareNextSlot.ts#L214

could further ratelimit it there in case we have multiple proposers in same epoch

Reply:
> What about running this prepare slot but only we are proposer, e.g. after emitting payload attributes

I've thought about this as well. But I don't like a periodic task like this depends on an event that is unpredictable and random. If we start bn and we fail to get EL info because EL starts after bn, then it could take months before the default graffiti is correctly set.

Reply:
> But I don't like a periodic task 

I wouldn't consider this a period task, we want the graffiti to be accurate once we do a proposal and  if we call it in the prepare phase it ensures it's always up-to-date

Reply:
> it takes ~4.5months to trigger the first update

even better, we only call the api when it actually needed or do I miss something?

Reply:
> even better, we only call the api when it actually needed or do I miss something?

Ah I see what you mean. I was thinking maybe this client info can be used for other purposes like debugging.

Lazy loading works if we just want it for block production.

Reply:
> I was thinking maybe this client info can be used for other purposes like debugging.

I guess it could but generally you know what EL you are running as an operator, and the implementation right now is pretty tailored towards having the information in the graffiti.

The execution engine could keep track of this information as well as it already aware of EL status
https://github.com/ChainSafe/lodestar/blob/b453b37d53a946b6faf73277fe1d75becff3ae8f/packages/beacon-node/src/execution/engine/http.ts#L100

Could only fetch version api if EL went online (in case of started late, or restart)